### PR TITLE
[7.0.x] service CIDR update

### PR DESCRIPTION
**Pull request by a-palchikov**
Forward port of https://github.com/gravitational/planet/pull/697.

### CODE REVIEW

File: tool/planet/agent.go [hunk 1]
```diff
@@ -481,6 +507,54 @@ func status(c statusConfig) (ok bool, err error) {
...
+			return trace.Wrap(err)
+		}
+		if svc.Spec.ClusterIP == "" {
+			// not a clusterIP service?
+			logger.Info("API server service clusterIP is empty.")
+			return nil
+		}
+		clusterIP := net.ParseIP(svc.Spec.ClusterIP)
+		// Kubernetes defaults to the first IP in service range for the API server service
+		// See https://github.com/kubernetes/kubernetes/blob/v1.15.12/pkg/master/services.go#L41
```
Comment (r0mant) [inline tool/planet/agent.go:532]: Should this link point to K8s 1.17 sources?

File: tool/planet/dns.go [hunk 2]
```diff
@@ -110,16 +107,50 @@ func updateEnvDNSAddresses(client *kubernetes.Clientset, role agent.Role) error
...
+			return nil
+		}
+		if isIPAlreadyAllocatedError(err) {
+			ipalloc.Release(ip)
+			ip, err = ipalloc.AllocateNext()
+			if err != nil {
+				return &backoff.PermanentError{Err: err}
+			}
+		}
+		logger.Warnf("Error creating service %v.", err)
```
Comment (r0mant) [inline tool/planet/dns.go:147]: WithError?

File: tool/planet/agent.go [hunk 3]
```diff
@@ -481,6 +507,54 @@ func status(c statusConfig) (ok bool, err error) {
...
+		return nil
+	})
+}
+
+func removeKubernetesService(services corev1.ServiceInterface) error {
+	err := services.Delete(KubernetesServiceName, &metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
+	return nil
```
Comment (r0mant) [inline tool/planet/agent.go:555]: Log not found case?

**Review by r0mant — APPROVED** [review]

File: tool/planet/agent.go (hunk 1 above)
Comment (a-palchikov) [reply tool/planet/agent.go:532]: Done

**Pull request merged**